### PR TITLE
bump: bump up for v1.2.0 release

### PR DESCRIPTION
**Pull request by Two-Hearts**

### CODE REVIEW

**Comment by Two-Hearts** [comment]
> Will we include #434 and #433 in this PR? or They will be another PR.

They are cherry picked and included in this PR.

**Review by JeyJeyGao — DISMISSED** [review]
LGTM

File: signer/signer.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ import (
 )
 
 // signingAgent is the unprotected header field used by signature.
-const signingAgent = "notation-go/1.2.0-rc.1"
```
Comment (shizhMSFT) [inline signer/signer.go:37]: Please split the PR for dep bump up and the vote.

Comment (Two-Hearts) [reply signer/signer.go:37]: This is notation-go library though, we are not voting from a PR. I think this update is more of a bump up?

Comment (Two-Hearts) [reply signer/signer.go:37]: As discussed, we will vote from a PR for notation-go library in the future.

**Review by JeyJeyGao — APPROVED** [review]
LGTM

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Review by priteshbandi — APPROVED** [review]
LGTM

**Pull request merged**